googletest/src/gtest.cc

```diff
@@ -734,13 +740,19 @@ class UnitTestFilter {
   // Constructs a filter from a string of patterns separated by `:`.
   explicit UnitTestFilter(const std::string& filter) {
     // By design "" filter matches "" string.
-    SplitString(filter, ':', &patterns_);
+    SplitString(filter, ':', &glob_patterns_);
```

Comment:
Instead of initializing `glob_patterns_` with all patterns and then erasing, could we use a local named `all_patterns` instead and then std::move respective ranges into `glob_patterns_` and `exact_match_patterns_` (the latter maybe using `std::inserter`)?